src/Skill/Progression.cpp

```diff
@@ -0,0 +1,67 @@
+//
+// Created by musique88 on 8/28/20.
+//
+
+#include <cmath>
+#include "Skill/Progression.h"
+
+#define FUNCTION_L 0.2f
+#define FUNCTION_CONSTANT_ADDED 20
+namespace CC
+{
+	int Progression::getLevel() const
+	{
+		return level_;
+	}
+
+// The function to get the needed xp for the next level is as follows:
+// f(level) = (level * l)^2
+	const int Progression::neededXpFunction(int level)
+	{
+		return std::floor((level * FUNCTION_L)*(level * FUNCTION_L)) + FUNCTION_CONSTANT_ADDED;
+	}
+//Leveling up adds one skillpoint to every skill
+	void Progression::levelUp()
+	{
+		level_++;
+	}
+
+	bool Progression::addXp(int xp)
+	{
+		xp_ += xp;
+		int neededXpToLevel = neededXpToLevelUp();
+		if (neededXpToLevel >= 0)
+		{
+			levelUp();
```

Comment:
Wait, if the amount of neededXpToLevel is positive, your level up? So if you need xp, you level up?

Reply:
Okay, I see that your code works... But why did you inverse the thing.

Reply:
done